src/hardware/setup.rs

```diff
@@ -341,13 +341,16 @@ pub fn setup(
                 polarity: hal::spi::Polarity::IdleHigh,
                 phase: hal::spi::Phase::CaptureOnSecondTransition,
             })
-            .manage_cs()
-            .suspend_when_inactive()
-            .communication_mode(hal::spi::CommunicationMode::Receiver)
-            .cs_delay(design_parameters::ADC_SETUP_TIME);
+            .inter_word_delay(design_parameters::ADC_SETUP_TIME)
+            .hardware_cs(hal::spi::HardwareCS {
+                mode: hal::spi::HardwareCSMode::WordTransaction,
+                assertion_delay: design_parameters::ADC_SETUP_TIME,
```

Comment:
Why is `ADC_SETUP_TIME` in there twice? There should only be the delay between asserting CS and the first bit (MSSI). MIDI should be 0.
If that duplication just a quirk of the new API, can we add a comment to clarify?

Reply:
No, MIDI should not be set to 0. We utilize MIDI because the SPI operates in endless transaction mode (TSIZE = 0), which means that we need to use MIDI as well as MSSI (MSSI is only used right as the transaction starts, then MIDI is used for all further words).

Reply:
However, upon probing the lines, it looks like ADC setup time is working even without this line. I'll look into this some more

Reply:
Ah, this is because my memory is failing me at an early age apparently. Upon reading docs around the ADCs, we actually removed the endless transaction mode in favor of explicitly clearing the EOT flag (I believe we had to to ensure setup timing). You're right that this line can be removed.